src/common/utils/usePagination.ts

```diff
@@ -1,21 +1,30 @@
 import { useHistory } from 'react-router-dom';
 import { useCallback } from 'react';
+import { History } from 'history';
 
 const PAGE_SIZE = 10;
 
 export const usePagination = (pageSize = PAGE_SIZE) => {
   const history = useHistory();
-
-  const urlSearchParams = new URLSearchParams(history.location.search);
-  const pageParam = Number(urlSearchParams.get('page') ?? 1);
+  const pageParam = getCurrentPageParam(history);
   const pageIndex = !Number.isNaN(pageParam) ? pageParam - 1 : 0;
   const cursorIndex = pageIndex * pageSize;
   const cursor = cursorIndex > 0 ? btoa(`arrayconnection:${cursorIndex - 1}`) : undefined;
 
   const getPageCount = (connectionsCount: number | null | undefined) =>
     connectionsCount ? Math.ceil(connectionsCount / pageSize) : 1;
 
-  const goToPage = useCallback((pageIndex: number) => history.push({ search: `?page=${pageIndex + 1}` }), [history]);
+  const goToPage = useCallback(
+    (pageIndex: number) => {
+      const newPageNumber = pageIndex + 1;
+      const currentPageNumber = getCurrentPageParam(history);
+
+      if (currentPageNumber !== newPageNumber) {
+        history.push({ search: `?page=${newPageNumber}` });
+      }
+    },
+    [history]
```

Comment:
According to [useCallback documentation](https://reactjs.org/docs/hooks-reference.html#usecallback):

>  every value referenced inside the callback should also appear in the dependencies array.

Reply:
I think that all the dependencies are listed in array, `history` is the only one. Others are just local consts.

Reply:
How about `getCurrentPageParam`?

Reply:
That one cannot change and it's a function, so it would make no difference. Linter is happy also: this is in compliance with `react-hooks/exhaustive-deps` rule.